test/forge/Strategies.sol

```diff
@@ -0,0 +1,1829 @@
+        vm.startPrank(user1);
+
+        Order memory order0 = generateTestOrder();
+        Order memory order1 = generateTestOrder();
+        if (order0Invalid) {
+            if (rateA) {
+                order0.A = 2 ** 64 - 1;
+            } else {
+                order0.B = 2 ** 64 - 1;
+            }
+        } else {
+            if (rateA) {
+                order1.A = 2 ** 64 - 1;
+            } else {
+                order1.B = 2 ** 64 - 1;
+            }
+        }
+
+        vm.expectRevert(Strategies.InvalidRate.selector);
+        carbonController.createStrategy(token0, token1, [order0, order1]);
+
+        vm.stopPrank();
+    }
+
+    function testStrategyCreationTokenSortingPersist(bool token0First) public {
+        vm.startPrank(user1);
+
+        Order memory order = generateTestOrder();
+        uint256 strategyId;
+        if (token0First) {
+            strategyId = carbonController.createStrategy(token0, token1, [order, order]);
+            Strategy memory strategy = carbonController.strategy(strategyId);
+            assertTrue(strategy.tokens[0] == token0);
+            assertTrue(strategy.tokens[1] == token1);
+        } else {
+            strategyId = carbonController.createStrategy(token1, token0, [order, order]);
+            Strategy memory strategy = carbonController.strategy(strategyId);
+            assertTrue(strategy.tokens[0] == token1);
+            assertTrue(strategy.tokens[1] == token0);
+        }
+
+        vm.stopPrank();
+    }
+
+    /**
+     * @dev strategy update tests
+     */
+
+    /// @dev test that the strategy update stores orders correctly
+    function testStrategyUpdateStoresOrdersCorrectly(uint256 i0, uint256 i1, int64[2] memory deltas) public {
+        vm.startPrank(user1);
+        // use two of the below 3 tokens for the strategy
+        Token[3] memory tokens = [token0, token1, NATIVE_TOKEN];
+        // pick two random numbers from 0 to 2 for the tokens
+        i0 = bound(i0, 0, 2);
+        i1 = bound(i1, 0, 2);
+        vm.assume(i0 != i1);
+        // bound deltas from -800000 to 800000
+        deltas[0] = int64(bound(deltas[0], -800000, 800000));
+        deltas[1] = int64(bound(deltas[1], -800000, 800000));
+
+        // generate order
+        Order memory order = generateTestOrder();
+
+        // create strategy
+        uint256 val = tokens[i0] == NATIVE_TOKEN || tokens[i1] == NATIVE_TOKEN ? order.y : 0;
+        uint256 strategyId = carbonController.createStrategy{ value: val }(tokens[i0], tokens[i1], [order, order]);
+
+        // create new orders
+        Order memory order0 = updateOrderDelta(order, deltas[0]);
+        Order memory order1 = updateOrderDelta(order, deltas[1]);
+
+        // update strategy
+        val = tokens[i0] == NATIVE_TOKEN && deltas[0] >= 0 ? abs(deltas[0]) : 0;
+        val += tokens[i1] == NATIVE_TOKEN && deltas[1] >= 0 ? abs(deltas[1]) : 0;
```

Comment:
You can get rid of the `abs` (you'll probably need to cast to uint though, but it would be safe to do so without using `safeCast`).

This code appears in many other places, so consider making it a custom helper function.

Reply:
Extracted the logic in a separate function, casted directly to uint64 instead of using abs.